mono/mini/interp/interp.c

```diff
@@ -1454,6 +1451,12 @@ interp_init_delegate (MonoDelegate *del)
 			mono_error_assert_ok (error);
 		}
 	}
+
+	if (!((InterpMethod *) del->interp_method)->transformed && method_is_dynamic (method)) {
```

Comment:
@lewurm What's the reasoning behind transforming the delegate method only if the method is dynamic, during delegate creation ?

Reply:
Good question. I observed two requirements:

1. If the (eventual) delegate body is part of the CIL image, the compilation error should be materialize as managed exception upon the first invocation. See `mono/tests/delegate15.cs`: https://github.com/mono/mono/blob/df37bb7a7c874fa33e1aa5ddf3648df4aec5d51e/mono/tests/delegate15.cs
2. If the delegate body is generated _at runtime_, such compilation errors are expected at delegate creation. See `MemberBindTests.cs`: https://github.com/mono/corefx/blob/68f67ad6ced8b74a65a1a5617a4c761bb203a25b/src/System.Linq.Expressions/tests/MemberInit/MemberBindTests.cs#L165-L178

This was previously covered by `create_delegate_method_ptr ()`: https://github.com/mono/mono/blob/85c8e85476fcc2076e3b6e362c77c62f463340c5/mono/mini/mini-runtime.c#L3555-L3570